File: 3514-number-of-unique-xor-triplets-ii/3514-number-of-unique-xor-triplets-ii.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var uniqueXorTriplets = function(nums) {
    const MAX = 2048;

    const pairXor = new Uint8Array(MAX);

    for (let i = 0; i < nums.length; i++) {
        for (let j = i; j < nums.length; j++) {
            pairXor[nums[i] ^ nums[j]] = 1;
        }
    }

    const ans = new Uint8Array(MAX);

    for (let x = 0; x < MAX; x++) {
        if (!pairXor[x]) continue;

        for (const num of nums) {
            ans[x ^ num] = 1;
        }
    }

    let count = 0;
    for (const v of ans) {
        if (v) count++;
    }

    return count;
};

File: leetcode/3514-number-of-unique-xor-triplets-ii/3514-number-of-unique-xor-triplets-ii.java
import java.util.BitSet;
import java.util.HashSet;
import java.util.Set;

class Solution {
    public int uniqueXorTriplets(int[] nums) {
        int n = nums.length;

        if (n == 1) {
            return 1;
        }

        Set<Integer> pairXor = new HashSet<>();

        for (int i = 0; i < n; i++) {
            for (int j = i + 1; j < n; j++) {
                pairXor.add(nums[i] ^ nums[j]);
            }
        }

        BitSet ans = new BitSet();

        for (int x : pairXor) {
            for (int num : nums) {
                ans.set(x ^ num);
            }
        }

        return ans.cardinality();
    }
}